controllers/topics/index.js

```diff
@@ -170,6 +170,52 @@ const getTopics = async (req, res) => {
   }
 };
 
+const createTopic = async (req, res) => {
+  try {
+    let {name} = req.body;
+    name = name.toLowerCase();
+
+    const {token, userId} = req;
+
+    let topics = await Topics.findOne({
+      where: {
+        name,
+        deleted_at: null
+      }
+    });
+
+    if (topics) {
+      return res.status(400).json({
+        code: 400,
+        status: 'failed',
+        message: 'Topic already exists'
+      });
+    }
+
+    const topic_id = await insertTopics([name]);
+    await followTopicViaCreateTopic({token, name, topic_id, userId});
+
+    return res.status(200).json({
+      success: true,
+      message: 'Success',
+      topic_id
+    });
+  } catch (error) {
+    return res.json({
+      code: error.statusCode,
+      status: 'fail',
+      message: error.message,
+      data: 'null'
+    });
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `createTopic` function seems to be doing a lot of things. It's checking if the topic exists, creating a new topic if it doesn't exist, and then following the topic. This is a violation of the Single Responsibility Principle (SRP) which states that a function should do one thing only. 

To improve this, you could separate the concerns into different functions. For example, you could have a `checkIfTopicExists` function, a `createNewTopic` function, and a `followTopic` function. This would make your code more modular, easier to read, and maintain.

Also, in line 204-209, when an error occurs, you are returning a response with a status code of 200 by default. This can be misleading as 200 usually indicates a successful operation. You should return an appropriate HTTP status code based on the type of error. For instance, for validation errors, you can return a 400 status code.

Here's how you could refactor the code:

```diff
const createTopic = async (req, res) => {
  try {
    let {name} = req.body;
    name = name.toLowerCase();

    const {token, userId} = req;

    let topics = await checkIfTopicExists(name);

    if (topics) {
      return res.status(400).json({
        code: 400,
        status: 'failed',
        message: 'Topic already exists'
      });
    }

    const topic_id = await createNewTopic(name);
    await followTopic({token, name, topic_id, userId});

    return res.status(200).json({
      success: true,
      message: 'Success',
      topic_id
    });
  } catch (error) {
    return res.status(error.statusCode || 500).json({
      code: error.statusCode,
      status: 'fail',
      message: error.message,
      data: 'null'
    });
  }
};
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->